src/core/auth/wallet-authenticator.ts

```diff
@@ -0,0 +1,342 @@
+
+    // Check that the intended domain matches the domain of the payload
+    if (payload.payload.domain !== domain) {
+      throw new Error(
+        `Expected domain ${domain} does not match domain on payload ${payload.payload.domain}`,
+      );
+    }
+
+    // Check that the payload hasn't expired
+    const currentTime = new Date();
+    if (currentTime > new Date(payload.payload.expirationTime)) {
+      throw new Error(`Login request has expired`);
+    }
+
+    // If chain ID is specified, check that it matches the chain ID of the signature
+    if (
+      parsedOptions?.chainId !== undefined &&
+      parsedOptions.chainId !== payload.payload.chainId
+    ) {
+      throw new Error(
+        `Chain ID ${parsedOptions.chainId} does not match payload chain ID ${payload.payload.chainId}`,
+      );
+    }
+
+    // Check that the signing address is the claimed wallet address
+    const message = this.generateMessage(payload.payload);
+    const userAddress = this.recoverAddress(message, payload.signature);
+    if (userAddress.toLowerCase() !== payload.payload.address.toLowerCase()) {
+      throw new Error(
+        `Signer address ${userAddress.toLowerCase()} does not match payload address ${payload.payload.address.toLowerCase()}`,
+      );
+    }
+
+    return userAddress;
+  }
+
+  /**
+   * Generate Authentication Token
+   * @remarks Server-side function that generates a JWT token from the provided login request that the
+   * client-side wallet can use to authenticate to the server-side application.
+   *
+   * @param domain - The domain of the server-side application to authenticate to
+   * @param payload - The login payload to authenticate with
+   * @param options - Optional configuration options for the authentication request
+   * @returns A authentication token that can be used by the client to make authenticated requests
+   *
+   * @example
+   * ```javascript
+   * const domain = "thirdweb.com";
+   * const loginPayload = await sdk.auth.login(domain);
+   *
+   * // Generate a JWT token that can be sent to the client-side wallet and used for authentication
+   * const token = await sdk.auth.generate(domain, loginPayload);
+   * ```
+   */
+  public async generate(
+    domain: string,
+    payload: LoginPayload,
+    options?: AuthenticationOptions,
+  ): Promise<string> {
+    if (isBrowser()) {
+      throw new Error(
+        "Authentication tokens should not be generated in the browser.",
+      );
+    }
+
+    const parsedOptions = AuthenticationOptionsSchema.parse(options);
+
+    const userAddress = this.verify(domain, payload);
+    const adminAddress = await this.wallet.getAddress();
+    const payloadData = AuthenticationPayloadDataSchema.parse({
+      iss: adminAddress,
+      sub: userAddress,
+      aud: domain,
+      nbf: parsedOptions?.invalidBefore || new Date(),
+      exp:
+        parsedOptions?.expirationTime ||
+        new Date(Date.now() + 1000 * 60 * 60 * 5),
```

Comment:
same question here, where does the 5h come from? Unsure what i would expect as a user if i don't pass expiration time, my first reaction would be - don't expire it. but maybe that's dangerous?

Reply:
Responded above on this + the other timing question - TL;DR the timings are important, 5m/5h are standard/acceptable for what we're using them for